### ISSUE TITLE
Ursache 9002 "file ist not a database, sqlite"

### ISSUE THREAD

**Issue by helmutweick** [corona-warn-app/cwa-app-android#579]
Next day after installation, data base seems corupted, over night without 
direct user inteference. 
App started new with setting activation and permissions. Then error message:
 "Ursache 9002
file ist not a database", while compiling select count(*) from sqlite_master"
Sorry, no detailed error due to lack of screenshot.

Result: cannot run app, crashes.
Can reinstall and run app but then contact data are also lost.
Huawei H60 with Android 6.0, Google Play Services 6.04, not rooted
Never had problems with file corruption.

![Unbenannt](https://user-images.githubusercontent.com/67044280/84870394-8f50e500-b07f-11ea-9fcb-c9f91b19dffb.JPG)

**Comment by helmutweick** [comment]
Issue was also mentioned for beta 11, but closed.

**Comment by tomjschwanke** [comment]
Contact details are not lost, they're stored by Google's API.

**Comment by jakobmoellersap** [comment]
Thanks for your issue! Could you give us a detailed explanation of what you did in the App and when you installed the App. Also it would be interesting to know if you have any security software running on your device that could potentially trigger a reinstatiation of the Android Master Key. Did you maybe do a device reset or anything in that direction that could trigger that?

**Comment by helmutweick** [comment]
Installed official version yesterday afternoon (16.06.), checked a few times to see how it behaves after cleaning cache. Next morning the issue, 
No reboot in this time, I do run Avast Virus scanner.

**Comment by jakobmoellersap** [comment]
Okay, thanks for the explanation. From our side, we cannot reproduce, unfortunately. We would kindly ask you to reinstall the application and see if the issue persists the day after installation (so that you can show that the issue persists).

**Comment by Ritzelprimpf** [comment]
Same Problem here:

![DSC_0525](https://user-images.githubusercontent.com/67065187/84919253-38640380-b0b1-11ea-9d02-ec57af14cb9a.JPG)
![DSC_0526](https://user-images.githubusercontent.com/67065187/84919260-3bf78a80-b0b1-11ea-9fb1-6e85693fef72.JPG)
![DSC_0527](https://user-images.githubusercontent.com/67065187/84919267-3dc14e00-b0b1-11ea-8681-206d2203af6b.JPG)
![DSC_0528](https://user-images.githubusercontent.com/67065187/84919271-3f8b1180-b0b1-11ea-8547-6f1545acd4d6.JPG)

**Comment by jakobmoellersap** [comment]
@Ritzelprimpf can you give context information regarding device, root (if applicable), installation context, installation duration and process of installation?

**Comment by jakobmoellersap** [comment]
Also, could both of you try the following:
1. Reinstall the App
2. Toggle the Exposure Notification switch a few times (as this triggers a database write)
3. Let the device run for a day
4. Check on the status again

**Comment by Ritzelprimpf** [comment]
Device: Honor 10
Android Version: 8.1.0
Installation was through PlayStore and took maybe a minute or so

Reinstalling the App does seem to fix the problem. It it fails again I will let you know.

**Comment by vvye** [comment]
Just reporting that I've been having the exact same problem (Huawei P8, Android 6.0). What I did was install the app, open it, and activate exposure logging - worked fine on the first run, when I opened it again a couple of hours later it started back at the welcome screen and eventually ran into the error.

Reopening the app did not fix it, but it looks like reinstalling did.

**Comment by jakobmoellersap** [comment]
Maybe for some Context: The only way this error occurs and that we have seen it occur, is if the read of the encrypted database failed. I am concerned that your devices alter the database file or had problems initializing the password during the first installation. If this is the case, we will have to gather more information about the context of installation as we cannot reproduce otherwise.

**Comment by helmutweick** [comment]
So far so good. After reinstallaton yesterday, today first status.
Inbetween I switched off and on the "Risiko-Emittlung" and the "Mitteilungen" a few times.
Switched on and off the phone and ran an Avast scan.

**Comment by jakobmoellersap** [comment]
Hi again,

After investigating more deeply into the issue, we cannot deduce a reliable error source and thus cannot reproduce. Unless we have more information regarding resolution of this issue, I will have to close this with the remark that a reinstall will fix the issue as a workaround.

I am sorry for the inconvenience, but since we have no dedicated crash reporting and can only rely on user feedback and device setups, this will have to be stalled until we have more information.

**Comment by corneliusroemer** [comment]
@jakobmoellersap  Maybe you should not close this as won't fix and just wait whether people can reproduce it with more information.

If you say you cannot reproduce, this may be because you're not using the same phone with the same OS version?

Also, the problems seem to have appeared only after some time - did you also wait for that long?

**Comment by corneliusroemer** [comment]
Isn't it pointless that this issue gets closed just to be reopened somewhere else because someone else encounters the same error?
#IssueClosurePolicy

**Comment by helmutweick** [comment]
Yes, it needs a few more days without error, before one can say that a new installaton is a satisfying solution. At least it seems to happen many times.

**Comment by jakobmoellersap** [comment]
You are correct. Please refer to my edited comment above.

**Comment by hannesa2** [comment]
This [android-database-sqlcipher](https://github.com/sqlcipher/android-database-sqlcipher) is crap, I observed it one year ago
https://github.com/sqlcipher/android-database-sqlcipher/issues/429
https://github.com/commonsguy/cwac-saferoom/issues/41
We used this in a app with installation > 5 millions and approximately 1.5 % of all users run into this error. We gave up and removed it

**Comment by vaubaehn** [comment]
Hi @helmutweick , @Ritzelprimpf , @vvye , it's quite a while since you reported this bug:
**"Ursache 9002 - File is not a database: ,while compiling: select count (*) from sqlite_master;"**

As far as I know, the root source for this exception is not known yet.
I came across the information, that the problem often occurs with Huawai and Honor devices.
One similar feauture of many of these devices is the PrivateSpace feature (Huawai/Honor specific privacy app & data storage container; see [here](https://consumer.huawei.com/de/support/content/de-de00746116/), [here](https://consumer.huawei.com/de/support/content/de-de00722628/), [here](https://www.giga.de/smartphones/huawei-mate-9/specials/top-10-tipps-zur-emui-5.0/page/2/), or [here](https://www.honornews.de/tweaks/privatespace-fur-mehr-privatsphaere/)).

Did you ever use or ever enable PrivateSpace for your phone, or might it have been preconfigured to be active when you bought it?

Are you still using CWA, with the newest version 1.2.0? If yes, is the error still persisting for you? Or is it gone?

A reply to these questions may help to get an idea, if PrivateSpace might be related to the issue you experienced.

Thanks very much in advance, kind regards,
V.

**Comment by vvye** [comment]
> Did you ever use or ever enable PrivateSpace for your phone, or might it have been preconfigured to be active when you bought it?
I hadn't heard of it before, and I can't find any mention of it on my device. If it's active by default, I don't know how to tell.

> Are you still using CWA, with the newest version 1.2.0? If yes, is the error still persisting for you? Or is it gone?
My current experience is this:
- CWA initially started working after some update in July.
- Since then, every couple of days the app stops working and seems to reset: when I open it, I get back to the welcome screen, and when I try to activate exposure logging again, I get the "file is not a database" error in question.
- Reinstalling the app makes it work again, at least for a few days until it happens again.
- This happened just a few minutes ago. The last time it happened before that was Saturday.

By "the app is working", I mean it displays my protection staus and tells me how long it has been active. I haven't gotten any warnings or any indication that things were being logged yet.

Thanks for investigating this!

**Comment by vaubaehn** [comment]
Hi @vvye , thanks for replying!
Sad to hear, that it still didn't work out for you. Some users in another issue thread reported, that updating to Android 9.1 or 10 solved the problem for them - but I guess, your P8 is out of support here?

Thanks to your feedback and of others, initial hints actually don't point to the feature PrivateSpace.

If the problems are solved with Android 9.1 or 10, my guess is, that upto Android 8 there was a special Java-implementation of security/OpenSSL from Huawei / Honor related Android, that was not complient to the standards.
If more reports point into that direction, one could ask to dev, if it is worth to have a deeper look at that, if show stopping differences may be found between implementations, and if it is possible to find a work around for that.

But let's wait some time for more feedback.

Kind regards, V.

**Comment by vaubaehn** [comment]
Hi @vvye , can you check if CWA is a 'geschützte App' for your device, like described here: https://github.com/corona-warn-app/cwa-app-android/issues/642#issuecomment-673567727
If not, please enable it as 'geschützt' and report, if it changed anything for you.
Thank you very much in advance!

**Comment by vaubaehn** [comment]
Hi @jakobmoellersap ,
can you try to trigger / reproduce the error by intentionally altering any random byte of the database file with a file monitor or simple editor?
If this produces the same stack trace, could it be in the scope of possibility, that the Huawei / Honor OS prior to Android 9.1 kill the task that handles the database, while it is open or even written to, for energy saving policies, and leave the database file corrupted on disk?

**Comment by vvye** [comment]
> Hi @vvye , can you check if CWA is a 'geschützte App' for your device, like described here: [#642 (comment)](https://github.com/corona-warn-app/cwa-app-android/issues/642#issuecomment-673567727)
> If not, please enable it as 'geschützt' and report, if it changed anything for you.
> Thank you very much in advance!
Just checked, and yes, it already is.

I have a tendency of completely closing apps by swiping them away on the list of open apps (if that makes sense), so maybe that's the problem? It doesn't seem likely since it wouldn't explain why the error comes up only every few days, but just to be sure I'll be keeping the app fully open now.

Thanks for being on it!

**Comment by vaubaehn** [comment]
Hi @vvye , I think it's definitively worth a try to leave it open and not swipe it away.
I'm suspecting that the database file may be corrupted by being killed by energy saving mechanisms of your device.
Maybe it can help as a first counter measure

**Comment by jnhmn** [comment]
I have the same issue on a Sony Xperia XZ2
Android 10 (Patch level 2020-07-01)
It just stopped working I'm not aware of any updates been installed. 

I switched to airplane mode several times, don't know if this is relevant.

**Comment by vaubaehn** [comment]
Hi @jnhmn 
> I have the same issue on a Sony Xperia XZ2
> I switched to airplane mode several times, don't know if this is relevant.

if your phone follows a restrictive energy saving policy when switching to airplane mode, then I'd say, yes, there may be a chance that it caused it. But just a guess, though.

I'd recommend to delete CWA's data and cache. If you don't have a pending COVID-test, you'd should not loose anything important by deleting these, just time of exposure logging being active is not accurate anymore. App should open properly after deleting cache and data.
Maybe you are able to reproduce the problem by switching to airplane mode and back again?

**Comment by hannesa2** [comment]
https://github.com/android/architecture-components-samples/pull/218#issuecomment-474610653 

https://github.com/commonsguy/cwac-saferoom uses the same https://github.com/sqlcipher/android-database-sqlcipher like you do
`implementation 'net.zetetic:android-database-sqlcipher:4.4.0'` 
and I observed the same errors with `4.4.0` as well.

**Comment by jnhmn** [comment]
@vaubaehn No, I wasn't affected by the bug mentioned in issue 597

Reinstalling the app solved my problem and since then it didn't occur a second time

**Comment by vaubaehn** [comment]
> 
> 
> @vaubaehn No, I wasn't affected by the bug mentioned in issue 597
> 
> Reinstalling the app solved my problem and since then it didn't occur a second time

@jnhmn thanks for your feedback!
